sympy/core/add.py

```diff
@@ -594,10 +594,12 @@ def _eval_is_positive(self):
     def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
-            c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
+            c, a = self.as_independent(*self.free_symbols)
```

Comment:
@smichr if we can improve ```as_independent()``` function then we can solve this issue. You think about another method ?